File: backend/src/main/java/org/example/backend/domain/member/entity/Member.java
package org.example.backend.domain.member.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDateTime;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long memberId;

    @Column(nullable = false, unique = true, length = 100)
    private String email;

    @Column(nullable = false, length = 255)
    private String password;

    @Column(unique = true, length = 50)
    private String nickname;

    @Column(length = 500)
    private String profileImage;

    private Integer tradeScore = 50;

    @CreatedDate
    private LocalDateTime createDate;

    @Column(nullable = false)
    private Long points = 0L;

    @Builder
    public Member(String email, String password, String nickname, String profileImage) {
        this.email = email;
        this.password = password;
        this.nickname = nickname;
        this.profileImage = profileImage;
    }

    public void updatePoints(Long newPoint) {
        this.points = newPoint;
    }

    public void updateMemberInfo(String email, String password, String nickname, String profileImage) {
        this.email = email;
        this.password = password;
        this.nickname = nickname;
        this.profileImage = profileImage;
    }
}

File: backend/src/main/java/org/example/backend/domain/member/entity/Member.kt
package org.example.backend.domain.member.entity

import jakarta.persistence.*
import org.springframework.data.annotation.CreatedDate
import java.time.LocalDateTime

@Entity
class Member(
    @Column(nullable = false, unique = true, length = 100)
    var email: String,

    @Column(nullable = false, length = 255)
    var password: String,

    @Column(unique = true, length = 50)
    var nickname: String,

    @Column(length = 500)
    var profileImage: String? = null

) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var memberId: Long? = null
        protected set

    var tradeScore: Int = 50
        private set

    @CreatedDate
    var createDate: LocalDateTime? = null
        protected set

    @Column(nullable = false)
    var points: Long = 0L

    fun updatePoints(newPoint: Long) {
        this.points = newPoint
    }

    fun updateMemberInfo(
        email: String,
        password: String,
        nickname: String,
        profileImage: String?
    ) {
        this.email = email
        this.password = password
        this.nickname = nickname
        this.profileImage = profileImage
    }

}
